src/components/common/lineChart.jsx

```diff
@@ -0,0 +1,149 @@
+import React, { Component } from 'react'
+import {
+  select,
+  scaleTime,
+  extent,
+  axisBottom,
+  scaleLinear,
+  max,
+  axisLeft,
+  line,
+  curveMonotoneX,
+  timeDay,
+  timeFormat,
+} from 'd3'
+import './lineChart.css'
+
+export default class LineChart extends Component {
+  state = {
+    data: [
+      {
+        date: new Date('2020-04-21'),
+        value: 0,
+      },
+      {
+        date: new Date('2020-04-22'),
+        value: 70,
+      },
+      {
+        date: new Date('2020-04-23'),
+        value: 90,
+      },
+      {
+        date: new Date('2020-04-24'),
+        value: 40,
+      },
+      {
+        date: new Date('2020-04-25'),
+        value: 90,
+      },
+      {
+        date: new Date('2020-04-26'),
+        value: 90,
+      },
+      {
+        date: new Date('2020-04-27'),
+        value: 90,
+      },
+    ],
+    marginTop: 10,
+    marginRight: 10,
+    marginBottom: 30,
+    marginLeft: 30,
+  }
+
+  render() {
+    const LABEL_PADDING = 5
+    const PADDING_LEFT = 20
+
+    const {
+      data,
+      marginTop,
+      marginRight,
+      marginBottom,
+      marginLeft,
+    } = this.state
+    const width = this.props.width - marginLeft - marginRight
+    const height = this.props.height - marginBottom - marginTop
+
+    let x = scaleTime()
+      .domain(
+        extent(data, function (d) {
+          return d.date
+        }),
+      )
+      .range([0, width])
+
+    let y = scaleLinear()
+      .domain([
+        0,
+        max(data, function (d) {
+          return +d.value
+        }),
+      ])
+      .range([height, 0])
+
+    const generateLine = line()
+      .x(function (d) {
+        return x(d.date)
+      })
+      .y(function (d) {
+        return y(d.value)
+      })
+      .curve(curveMonotoneX)
```

Comment:
Not sure about d3, but does these variables need to be in render? Because they are gonna be calculated in every render, although the graph is gonna be pretty static (even when we get API data).

Reply:
Temporary fix since that component is not re-rendered. Will use lifecycle methods when making it dynamic.